Subject: dm:check_device_area fails on DASD devices during table create
From: Mike Anderson <andmike@us.ibm.com>
References: 158777 - LTC22470

Allow check_device_area to succeed if a device has an i_size of zero. This
addresses an issue seen on DASD devices setting up a multipath table for
paths in online and offline state.

Signed-off-by: Mike Anderson <andmike@us.ibm.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>

 drivers/md/dm-table.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletion(-)

Index: linux-2.6.16-rc5-git9-2/drivers/md/dm-table.c
===================================================================
--- linux-2.6.16-rc5-git9-2.orig/drivers/md/dm-table.c	2006-03-09 22:28:47.000000000 -0800
+++ linux-2.6.16-rc5-git9-2/drivers/md/dm-table.c	2006-03-16 10:18:03.000000000 -0800
@@ -389,7 +389,8 @@ static int check_device_area(struct dm_d
 {
 	sector_t dev_size;
 	dev_size = dd->bdev->bd_inode->i_size >> SECTOR_SHIFT;
-	return ((start < dev_size) && (len <= (dev_size - start)));
+	return dev_size ?
+		((start < dev_size) && (len <= (dev_size - start))) : 1;
 }
 
 /*
